### PR TITLE
CY-1052 Add initial commit for bootable volume for server blueprint

### PR DESCRIPTION
**Pull request by mabuaisha**

### CODE REVIEW

**Review by EarthmanT — APPROVED** [review]

File: openstack_sdk/resources/compute.py [hunk 1]
```diff
@@ -134,6 +134,46 @@ def get_server_password(self):
             ' password for this server: {0}'.format(server))
         return self.connection.compute.get_server_password(server)
 
+    def list_volume_attachments(self, query=None):
+        query = query or {}
+        self.logger.debug('Attempting to list servers')
```
Comment (EarthmanT) [inline openstack_sdk/resources/compute.py]: Attempting to list **volumes attachments**

File: openstacksdk_plugin/utils.py [hunk 2]
```diff
@@ -605,3 +597,17 @@ def merge_resource_config(resource_config, config):
...
+    This method helps to generate attachment volume key which can be used as
+    runtime property when running attaching/detaching volume from/to server
+    :param str prefix: Any prefix that could be added to the the key
+    :param str volume_id: Unique volume id
+    :param str server_id: Unique server id
+    :return str: attachment volume key
+    """
+    if all([prefix, volume_id, server_id]):
+        return '{0}-{1}-{2}'.format(prefix, volume_id, server_id)
+    return None
```
Comment (EarthmanT) [inline openstacksdk_plugin/utils.py]: What happens if `None` is received.

File: plugin.yaml [hunk 3]
```diff
@@ -2,9 +2,9 @@ plugins:
 
   openstacksdk:
     executor: central_deployment_agent
-    source: https://github.com/cloudify-incubator/cloudify-openstacksdk-plugin/archive/dev.zip
+    source: https://github.com/cloudify-incubator/cloudify-openstacksdk-plugin/archive/CY-1052-add-support-for-attaching-volume.zip
```
Comment (EarthmanT) [inline plugin.yaml]: need to update before merge.

**Review by EarthmanT — APPROVED** [review]
minor changes

File: openstack_sdk/resources/compute.py (hunk 1 above)
Comment (mabuaisha) [reply openstack_sdk/resources/compute.py]: Done

File: plugin.yaml (hunk 3 above)
Comment (mabuaisha) [reply plugin.yaml]: Yeah sure, this is just for testing

File: openstacksdk_plugin/utils.py (hunk 2 above)
Comment (mabuaisha) [reply openstacksdk_plugin/utils.py]: I updated it so that we can fallback to default key

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**